server/src/main/java/org/apache/iotdb/db/query/control/QueryResourceManager.java

```diff
@@ -85,20 +110,50 @@ public void registerTempExternalSortFile(long queryId,
     externalSortFileMap.computeIfAbsent(queryId, x -> new ArrayList<>()).add(deserializer);
   }
 
-
   public QueryDataSource getQueryDataSource(Path selectedPath,
       QueryContext context, Filter filter) throws StorageEngineException, QueryProcessException {
 
     SingleSeriesExpression singleSeriesExpression = new SingleSeriesExpression(selectedPath,
         filter);
-    return StorageEngine.getInstance().query(singleSeriesExpression, context, filePathsManager);
+    QueryDataSource queryDataSource = StorageEngine.getInstance()
+        .query(singleSeriesExpression, context, filePathsManager);
+    // calculate the distinct number of seq and unseq tsfiles
+    if (config.isEnablePerformanceTracing()) {
+      seqFileNumMap.computeIfAbsent(context.getQueryId(), k -> new HashSet<>())
+          .addAll(queryDataSource.getSeqResources());

```

Comment:
Hi, the TsFileResources used for this query in this map will be removed after printing tracing log. Then it can be cleaned by GC.

Reply:
You can see the Code in QueryResourceManager 136-146.

Reply:
this method just query one time series.. 
Suppose we have 2 time series to be queried, maybe the resources corresponding to the first time series can be GC released before the second one is queried.

Reply:
But if the first time series is not kept in this map, we can not calculate the total number of distinct tsfiles in the whole query. Or we give the accessed times of tsfile instead? In this way, we only keep a accessed number for one time series.

Reply:
Hi, I replace those objects with WeakReference as you said. I believe it will make sense in the GC. Thank you very much for your good suggestion. Please check it.

Reply:
And it will reflect the correct performance in case the tsfileResources are GC and reloaded from disk in the future.